illumos-utils/src/svcs.rs

```diff
@@ -0,0 +1,331 @@
+        let cmd = cmd.args(&[SVCS, "-Za", "-H", "-o", "state,fmri,zone"]);
+        let output = execute_async(cmd).await?;
+        let svcs_result = SvcsInMaintenanceResult {
+            services: SvcInMaintenance::parse(log, &output.stdout),
+            time_of_status: Some(Utc::now()),
+        };
+        Ok(svcs_result)
+    }
+
+    #[cfg(not(target_os = "illumos"))]
+    pub async fn in_maintenance(
+        log: &Logger,
+    ) -> Result<SvcsInMaintenanceResult, ExecutionError> {
+        info!(log, "OS not illumos, will not check state of SMF services");
+        let svcs_result = SvcsInMaintenanceResult::new();
+        Ok(svcs_result)
+    }
+}
+
+/// Lists services in maintenance status if any, and the time the health check
+/// for SMF services ran
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct SvcsInMaintenanceResult {
+    pub services: Vec<SvcInMaintenance>,
+    pub time_of_status: Option<DateTime<Utc>>,
+}
+
+impl SvcsInMaintenanceResult {
+    pub fn new() -> Self {
+        Self { services: vec![], time_of_status: None }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.services.is_empty() && self.time_of_status == None
+    }
+}
+
+/// Each service instance is always in a well-defined state based on its
+/// dependencies, the results of the execution of its methods, and its potential
+/// contracts events. See <https://illumos.org/man/7/smf> for more information.
+#[derive(
+    Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, JsonSchema,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum SvcState {
+    /// Initial state for all service instances.
+    Uninitailized,
+    /// The instance is enabled, but not yet running or available to run.
+    Offline,
+    /// The instance is enabled and running or is available to run.
+    Online,
+    /// The instance is enabled and running or available to run. It is, however,
+    /// functioning at a limited capacity in comparison to normal operation.
+    Degraded,
+    /// The instance is enabled, but not able to run.
+    Maintenance,
+    /// The instance is disabled.
+    Disabled,
+    /// Represents a legacy instance that is not managed by the service
+    /// management facility.
+    LegacyRun,
+    /// We were unable to determine the state of the service instance.
+    Unknown,
+}
+
+impl From<String> for SvcState {
+    fn from(value: String) -> Self {
+        match value.as_str() {
+            "Uninitailized" | "uninitailized" => SvcState::Uninitailized,
+            "Offline" | "offline" => SvcState::Offline,
+            "Online" | "online" => SvcState::Online,
+            "Degraded" | "degraded" => SvcState::Degraded,
+            "Maintenance" | "maintenance" => SvcState::Maintenance,
+            "Disabled" | "disabled" => SvcState::Disabled,
+            "Legacy Run" | "legacy run" | "Legacy run" | "legacy_run"
+            | "legacy-run" => SvcState::LegacyRun,
```

Comment:
Why do we have all these different cases?  Isn't the tool consistent in how it reports these?

Reply:
It is, but I always worry that the output will change in a new version or whatever, and then we have a silly parsing bug. Is there much harm in keeping these different cases?